pmlc/dialect/stripe/from_mlir.cc

```diff
@@ -65,6 +66,61 @@ class StripeBuilder {
   bool found_inst_;
 };
 
+struct StripeDevice {
+  std::string name;
+  std::vector<AffinePolynomial> units;
+};
+
+struct StripeLocation {
+  std::vector<StripeDevice> devs;
+};
+
+std::pair<FlatTensorAccess, StripeLocation> ComputeAccessAndLoc(Value* tensor) {
+  auto ret = ComputeAccess(tensor);
+  auto loc = StripeLocation{};
+
+  // At this point: ret.access contains the refinement accessors for every tensor dimension (including
+  // non-address dimensions), and ret.base_type is a TensorType which gets us the underyling dimension
+  // information.  For conversion to Stripe, we want to separate out the address dimensions (whose accessors
+  // are preserved in the flat tensor access) from the non-address dimensions (which become part of the
+  // location).
+
+  std::vector<AffinePolynomial> access;
+
+  auto combIt = ret.access.begin();
+  auto matches = llvm::SmallVector<StringRef, 4>();
+  for (const auto& dim : ret.base_type.getShape()) {
+    if (combIt == ret.access.end()) {
+      break;  // Should never happen; this is just to be careful.
+    }
+    if (dim.cls == "address") {
```

Comment:
`"address"` seems special, can we make a `const char* kDefaultClass = "address";` someplace?

Reply:
Done